common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -266,12 +271,37 @@ def destroy(self, request, uuid):
                 data='The Entitlement for this UUID does not exist or is Expired.'
             )
 
-        if entitlement.enrollment_course_run is None:
-            return Response(status=status.HTTP_204_NO_CONTENT)
+        with transaction.atomic():
```

Comment:
Just a question why there is a transaction here? It looks like there's only one update in this block

Reply:
Nevermind figured it out, _unenroll calls set_enrollment which has a .save() in it